### ISSUE TITLE
Is d3.chart still being developed/maintained/used?

### ISSUE THREAD

**Issue by joshuarrrr** [misoproject/d3.chart#115]
I've found it to be a useful tool, but it seems like activity and interest are declining

**Comment by kdh6429** [comment]
agree, but still some library use this framework or some bugs are fixed sometimes.

**Comment by andreas-marschke** [comment]
@joshuarrrr @kdh6429 My observation is that since d3.chart does not provide anything that needs frequent updates and since it's basically just an OO layer ontop of d3 its fine as it is and there is nothing much to do here. That is unless d3 would change dramatically which I can't see happening any time soon.

**Comment by mgerring** [comment]
I'm using it heavily and I'm starting to get a little worried -- there are a lot of outstanding, unresolved issues with it and if it's not under active development, I feel like forking it might be a good move.

**Comment by jugglinmike** [comment]
Hey everybody!

@iros and myself (the current project maintainers) are committed to making sure d3.chart is functioning correctly. Many of the open issues are feature requests. For the most part, these have stalled due to lack of interest/direction from the requester. If anyone feels that something is lacking in this library, we're always happy to provide guidance. Feature work requires a champion, though; new features are far more likely to land when they have a strong motivation and a viable implementation.

So when it comes to the relatively low amount of recent activity, I'm inclined to agree with @andreas-marschke. We continue to use d3.chart for production-ready projects ([Stereotropes](http://stereotropes.bocoup.com) being the most recent example), and we're still finding that it serves its purpose as a thin abstraction around the powerful D3.js.

As for forking the library, thanks to [the project's permissive license](https://github.com/misoproject/d3.chart/blob/4e55cc67a77979d4a040ec18e3b9cd6303c5d6ad/LICENSE-MIT), that is anyone's right. Actually, there is already one fork of d3.chart (that we're aware of): [Koto](https://github.com/nicksrandall/kotojs). That said, forking an open source project tends to lead to duplicated efforts and ecosystem fragmentation. Sometimes this has to be accepted as an unfortunate consequence of more serious problems: uncooperative maintainers, fundamental differences in vision, etc. So we do request that anyone considering this exercises some restraint before maintaining their own fork. If you'd like to see something changed, please start by bringing it up here!

**Comment by mgerring** [comment]
We're using d3.chart pretty heavily at [Genability](http://www.genability.com), and many of the things we ran into and had to implement ourselves are also covered by Koto. Namely:

*Common configuration API
*Data accessor API
*Extra hooks (preDraw and postDraw)

I think many of those things have come up in this issues section as well, so I think those three things especially are great candidates for inclusion in d3.chart.

**Comment by girlwithglasses** [comment]
@mgerring Are your modifications (common config API, extra hooks, etc.) available on GitHub or another open source repo?

**Comment by nicksrandall** [comment]
@girlwithglasses The best way to implement those features using D3.chart would be to create your own "base chart" that implements those features and then have all your components extend the base chart. I did that for a long time before I built KotoJS and it worked really well.
